### PR TITLE
[RELENG-7490] ✨ Adding GitHub Auth

### PR DESCRIPTION
**Pull request by gaspardmoindrot**

### CODE REVIEW

File: gh_actions_exporter/main.py [hunk 1]
```diff
@@ -20,6 +21,11 @@ def metrics() -> Metrics:
     return Metrics(get_settings())
 
 
+@lru_cache()
+def github_client() -> GithubClient:
+    return GithubClient(get_settings())
+
+
```
Comment (tcarmet) [inline gh_actions_exporter/main.py:28]: Do we need to add a class to return the github client? I feel like the GitHub class is already the one we want.
```suggestion
@lru_cache()
def github_client() -> GitHub:
    settings: Settings = get_settings()

    return GitHub(
        AppInstallationAuthStrategy(
            settings.github_app_id,
            settings.github_app_private_key.get_secret_value(),
            settings.github_app_installation_id
        )
    )
```

**Review by tcarmet — CHANGES_REQUESTED** [review]

File: gh_actions_exporter/main.py (hunk 1 above)
Comment (gaspardmoindrot) [reply gh_actions_exporter/main.py:28]: You're right! I change this

**Review by tcarmet — APPROVED** [review]
LGTM

you might need to merge this PR #31, rebase so that the settings are optional and the tests pass

File: gh_actions_exporter/config.py [hunk 2]
```diff
@@ -46,6 +46,8 @@ class Settings(BaseSettings):
     flavor_label: Optional[str] = "flavor"
     default_cost: Optional[float] = 0.008
 
+    exporter_enabled: bool = True
```
Comment (tcarmet) [inline gh_actions_exporter/config.py:49]: Either remove this var for now, or implement it

File: gh_actions_exporter/main.py [hunk 3]
```diff
@@ -20,6 +21,20 @@ def metrics() -> Metrics:
     return Metrics(get_settings())
 
 
+@lru_cache()
+def github_client() -> GitHub:
```
Comment (tcarmet) [inline gh_actions_exporter/main.py:25]:
```suggestion
def github_client() -> GitHub or None:
```

**Review by tcarmet — CHANGES_REQUESTED** [review]

**Review by tcarmet — APPROVED** [review]

**Pull request merged**